packages/keystone/redis.spec.js

```diff
@@ -0,0 +1,82 @@
+const Redis = require('ioredis')
+
+const conf = require('@open-condo/config')
+
+
+describe('Redis adapter', () => {
+    const OLD_ENV = process.env
+    let redisClient
+    let nonPrefixedClient
+
+    beforeAll(() => {
+        jest.resetModules()
+        process.env.REDIS_URL = conf['REDIS_URL'] || 'redis://127.0.0.1:6379'
+        process.env.REDIS_FALLBACK_ENABLED = conf['REDIS_FALLBACK_ENABLED'] || 'true'
+
+        const { getRedisClient } = require('./redis')
+        redisClient = getRedisClient('test')
+        nonPrefixedClient = new Redis(process.env.REDIS_URL)
+    })
+
+    afterAll(async () => {
+        await nonPrefixedClient.del('test')
+        await nonPrefixedClient.del('keystone:test1')
+        await nonPrefixedClient.del('keystone:incrTest')
+        await nonPrefixedClient.del('keystone:testList')
+        await nonPrefixedClient.disconnect()
+        await redisClient.disconnect()
+
+        process.env = { ...OLD_ENV }
```

Comment:
This will not work:
<img width="741" alt="image" src="https://github.com/user-attachments/assets/d5adfb9b-7bc2-4625-92af-5fcac4ed67fd" />

Since `process.env` is object, you're only creating new reference in OLD_ENV in line 7. So now you have 2 pointers, but same object

So then you're mutating object by adding new property (`process.env.something = 'something` is equal to `OLD_ENV.something = 'something'`, since pointers are same)

And after all you're spreading here not old copy, but mutated one

Reply:
Ok, fixed that with deep copy